core/src/main/scala/kafka/log/FileMessageSet.scala

```diff
@@ -117,7 +117,13 @@ class FileMessageSet private[kafka](@volatile var file: File,
     new FileMessageSet(file,
                        channel,
                        start = this.start + position,
-                       end = math.min(this.start + position + size, sizeInBytes()))
+                       end = {
+                         // Handle the integer overflow
+                         if (this.start + position + size < 0)
```

Comment:
The assumption here is that ```this.start + position``` is a positive integer, and ```size``` is also a positive integer. When you say not reliable, do you mean ```this.start + position``` can also overflow?